### ISSUE TITLE
How to communicate non-official bindists

### ISSUE THREAD

**Issue by hasufell** [haskell/ghcup-hs#789]
GHCup currently does provide non-official bindists for various tools. There can be good reasons for this, however, we currently have no nice way of communicating this to the user.

As a side-note: we provide an alternative [vanilla](https://github.com/haskell/ghcup-metadata/blob/develop/ghcup-vanilla-0.0.7.yaml) bindist metadata, that does not include any non-official bindists.

Some ideas:

* just communicate this at the end of the boostrap installation process with a link to documentation explaining the GHCup policies and alternatives (like the vanilla metadata)
* have a way to communicate this via `ghcup tui` or `ghcup list` (how?)

@mpickering @bgamari @david-christiansen

**Comment by hasufell** [comment]
Another point is (and my personal opinion): most end-users probably won't care about official vs non-official, but they probably care whether the *test suite passes* (as in: QA).

**Comment by chreekat** [comment]
Is this the place to argue against unofficial bindists? :D 

End-users probably also want *support* for the tools that GHCUp provides. It would be unfair to expect upstream to provide that support. Does GHCUp intend to provide it? 

I also think it's hard to draw the line between "unofficial" and "official" when GHCUp is, in fact, as "official" a tool as any. It's  confusing to have an official tool say it's doing something unofficial. What does that mean?

**Comment by hasufell** [comment]
> End-users probably also want support for the tools that GHCUp provides.

This is the same as everywhere. 

You're probably running a Debian distro somewhere, either on Desktop or a Server.

They have their own GHC package and may mess with the build system or even the source code.

What type of support can you expect from Debian maintainers if things go wrong? What type of support can you get from GHC developers wrt unsupported debian configuration? 

The same answers apply to GHCup.

> It would be unfair to expect upstream to provide that support.

I can turn this around and say: isn't it unfair that GHCup has to fix bindists downstream at times or build bindists for which upstream decided they got no time for anymore? 

I don't know what's fair in open source. I get bug reports about everything: GHC, cabal, HLS, VSCode,. ..

It probably is fair to communicate that unofficial bindists mean: you might get little or no upstream support. And only non-engineering support from GHCup.

> It's confusing to have an official tool say it's doing something unofficial. What does that mean?

It means it's using bindists that were not prepared by upstream.

**Comment by hasufell** [comment]
> I know most distro patches (perhaps not all) are added in order to improve compat with the rest of the distro, and I hope they are considered a necessary evil

Not really. Distros have vastly different policies. Gentoo used to carry heavy kernel patches and then introduced another vanilla-kernel package.

A debian maintainer mindlessly patched OpenSSL, because of incompetency and caused a major security incidence: https://www.gergely.risko.hu/debian-dsa1571.en.html

Some distros just pick up patches here and there for whatever reason. Some distros have strict policies to e.g. report patches to upstream first. It depends. Debian is surely one of the worst citizens here.

My point is: GHCup is no different here. It's a distribution channel like Debian, but with different policies. All the same caveats apply as for any distribution channel. There's nothing new. This is how it's been for decades. You can read about GHCup's distribution policies here: https://www.haskell.org/ghcup/about/#distribution-policies

> The GHC Platform Policy identifies the solution here: if someone wants to step up to maintain support for a platform, GHC HQ will welcome them and promote the platforms to Tier 1. It sounds like you do want to main support for certain platforms. If something prevented that from happening, I think it would be good to figure out what it was and deal with it.

How do I support a platform for which no gitlab CI infrastructure exists?
